pkg/daemon/controller.go

```diff
@@ -181,12 +177,69 @@ func (c *Controller) processNextSubnetWorkItem() bool {
 	return true
 }
 
-func (c *Controller) reconcileRouters() error {
+func (c *Controller) reconcileRouters(event subnetEvent) error {
 	subnets, err := c.subnetsLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("failed to list namespace %v", err)
 		return err
 	}
+
+	var ok bool
+	var oldSubnet, newSubnet *kubeovnv1.Subnet
+	if event.old != nil {
+		if oldSubnet, ok = event.old.(*kubeovnv1.Subnet); !ok {
+			klog.Errorf("expected old subnet in subnetEvent but got %#v", event.old)
+			return nil
+		}
+	}
+	if event.new != nil {
+		if newSubnet, ok = event.new.(*kubeovnv1.Subnet); !ok {
+			klog.Errorf("expected new subnet in subnetEvent but got %#v", event.new)
+			return nil
+		}
+	}
+
+	// handle policy routing
+	rulesToAdd, rulesToDel, routesToAdd, routesToDel, err := c.diffPolicyRouting(oldSubnet, newSubnet)
```

Comment:
why not just compare to exist routes and rules in the host?

Reply:
In case the controller deletes rules or routes which are created by users or by other processes.

Reply:
> why not just compare to exist routes and rules in the host?

@oilbeater In case the controller deletes rules or routes which are created by users or by other processes.